File: lang_py/Cpu.py
from ProcessingUnit import ProcessingUnit

class Cpu(ProcessingUnit):

    def __init__(self, merk, nama, jumlahCore, kecepatanGHz, tipeUnit, cacheSizeMB):
        super().__init__(merk, nama, jumlahCore, kecepatanGHz, tipeUnit)
        self.cacheSizeMB = cacheSizeMB

    def setCacheSizeMB(self, cacheSizeMB):
        self.cacheSizeMB = cacheSizeMB

    def getCacheSizeMB(self):
        return self.cacheSizeMB
    
    

File: lang_java/Cpu.java
class Cpu extends ProcessingUnit{

    private int cacheSizeMB;

    public Cpu(){
    }

    public Cpu(String merk, String nama, int jumlahCore, double kecepatanGHz, String tipeUnit, int cacheSizeMB){
        super(merk, nama, jumlahCore, kecepatanGHz, tipeUnit);
        this.cacheSizeMB = cacheSizeMB;
    }

    public void setCacheSizeMB(int cacheSizeMB){
        this.cacheSizeMB = cacheSizeMB;
    }

    public int getCacheSizeMB(){
        return cacheSizeMB;
    }

}